pkg/scheduler/reconciliation/filter.go

```diff
@@ -23,6 +26,43 @@ func (ws *WithSchedulingID) FilterByInstance(i *model.ReconciliationEntity) *mod
 	return nil
 }
 
+type WithRuntimeIDs struct {
+	RuntimeIDs []string
+}
+
+func (wc *WithRuntimeIDs) FilterByQuery(q *db.Select) error {
+	runtimeIDsLen := len(wc.RuntimeIDs)
+	if runtimeIDsLen < 1 {
+		return nil
+	}
+
+	var values string
+	for i := range wc.RuntimeIDs {
+		if i == runtimeIDsLen-1 {
+			values = fmt.Sprintf("%s$%d", values, i+1)
+			break
+		}
+		values = fmt.Sprintf("%s$%d,", values, i+1)
+	}
+
+	q.WhereIn("RuntimeID", values, strings.Join(wc.RuntimeIDs, ","))
```

Comment:
I mean it expected the values of the placeholders as array, concatenation isn't required actually

Reply:
This is the tricky one.
The `WhereIn` method expects arguments as `args ... interface{}` and the `wc.RuntimeIDs` type is `[]string`.
[If we pass the string slice we will pass it as `interface{}`](https://play.golang.org/p/vQaKSXLPh03) (we cannot spread it because the types do not match `[]interface` vs `[]string`).

Reply:
oh, that's quite hacky :D 

Can't we use something like this?

     q.WhereIn("RuntimeID", values, func (args []string) []interface{} {
        var result []interface{}
        for _, arg := range args {
           result = append(result, arg)
        }
        return result
     }(wc.RuntimeIDs)...)

Reply:
Let's do the conversion. I will just create dedicated function to avoid confusion with immediate executed lambda.